src/L2/paused/L2LockingPositionPaused.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.23;
+
+import { L2LockingPosition } from "src/L2/L2LockingPosition.sol";
+
+/// @title L2LockingPositionPaused - Paused version of L2LockingPosition contract
+/// @notice This contract is used to pause the L2LockingPosition contract. In case of any emergency, the owner can
+///         upgrade and pause the contract to prevent any further staking operations.
+contract L2LockingPositionPaused is L2LockingPosition {
+    error LockingPositionIsPaused();
+
+    /// @notice Setting global params.
+    function initializePaused() public reinitializer(2) { }
+
+    /// @notice Override the transferFrom function to prevent staking from being processed.
+    function transferFrom(address, address, uint256) public virtual override {
+        revert LockingPositionIsPaused();
+    }
+
+    /// @notice Override the createLockingPosition function to prevent staking from being processed.
+    function createLockingPosition(
+        address,
+        address,
+        uint256,
+        uint256
+    )
+        external
+        virtual
+        override
+        onlyStaking
```

Comment:
For complex custom modifiers I would agree, yes!
I can only see a danger if someone wants to use this contract as a baseline for a fixed version because a developer could miss that he needs to add this custom modifier back when implementing a fix version of a function.

So I would say to remove this, a quite hard to discover bug may be introduced.

Reply:
valid point. Thanks for the feedback.